src/pkg/reg/adapter/volcenginecr/adapter.go

```diff
@@ -0,0 +1,153 @@
+package volcenginecr
+
+import (
+	"errors"
+	"path"
+	"strings"
+
+	"github.com/goharbor/harbor/src/lib/log"
+	adp "github.com/goharbor/harbor/src/pkg/reg/adapter"
+	"github.com/goharbor/harbor/src/pkg/reg/adapter/native"
+	"github.com/goharbor/harbor/src/pkg/reg/model"
+	"github.com/goharbor/harbor/src/pkg/reg/util"
+	"github.com/goharbor/harbor/src/pkg/registry/auth/bearer"
+	volcCR "github.com/volcengine/volcengine-go-sdk/service/cr"
+	"github.com/volcengine/volcengine-go-sdk/volcengine"
+	"github.com/volcengine/volcengine-go-sdk/volcengine/credentials"
+	volcSession "github.com/volcengine/volcengine-go-sdk/volcengine/session"
+)
+
+func init() {
+	if err := adp.RegisterFactory(model.RegistryTypeVolcCR, new(factory)); err != nil {
+		log.Errorf("failed to register factory for %s: %v", model.RegistryTypeVolcCR, err)
+		return
+	}
+	log.Infof("the factory for adapter %s registered", model.RegistryTypeVolcCR)
+}
+
+type factory struct{}
+
+/**
+	* Implement Factory Interface
+**/
+var _ adp.Factory = &factory{}
+
+type adapter struct {
+	*native.Adapter
+	registryName *string
+	volcCrClient *volcCR.CR
+	registry     *model.Registry
+}
+
+// Create ...
+func (f *factory) Create(r *model.Registry) (adp.Adapter, error) {
+	return newAdapter(r)
+}
+
+// AdapterPattern ...
+func (f *factory) AdapterPattern() *model.AdapterPattern {
+	return getAdapterInfo()
+}
+
+func getAdapterInfo() *model.AdapterPattern {
+	return &model.AdapterPattern{}
+}
+
+func newAdapter(registry *model.Registry) (a *adapter, err error) {
+	// get region and registryName from url
+	region, registryName, err := getRegionRegistryName(registry.URL)
+	if err != nil {
+		log.Errorf("[volcengine-cr.newAdapter] getRegion failed. error=%v", err)
```

Comment:
Because the log in Harbor already include the source code line information, please remove the `[volcengine-cr.newAdapter]` from log to keep the same log style with the existing code.

Reply:
ok

> Because the log in Harbor already include the source code line information, please remove the `[volcengine-cr.newAdapter]` from log to keep the same log style with the existing code.

ok